MantidQt/CustomInterfaces/src/Reflectometry/QtReflRunsTabView.cpp

```diff
@@ -46,13 +46,13 @@ void QtReflMainView::initLayout() {
   ui.splitterTables->setStretchFactor(0, 0);
   ui.splitterTables->setStretchFactor(1, 1);
 
+  // Create the DataProcessor presenter
   ReflGenericDataProcessorPresenterFactory presenterFactory;
```

Comment:
The best solution (most flexible) will be to ultimately push the creation/decision of the concrete factories right up to the outside of your top-most view. So that you can do something like this:
```cpp
new QtReflMainViewWindow(new StarndardISISPresenterFactory{}); 
```

Where you have an `AbstractFactory` (read AbstractFactory pattern)

```cpp
class AbstractPresenterFactory{

virtual IReflGenericDataProcessorPresenter* createDataProcessorPresenter() = 0;

virtual IReflRunTabPresenter* createRunTabPresenter() = 0;

...

};

class StarndardISISPresenterFactory : public AbstractPresenterFactory {
// Implement all concrete presenter types here.
};

```
At this point this is a minor problem to solve, so I wouldn't worry about changing what you have if you are not convinced.